pkg/cloud-provider/cloudapi/aws/aws_ec2.go

```diff
@@ -210,30 +233,32 @@ func (ec2Cfg *ec2ServiceConfig) getInstances() ([]*ec2.Instance, error) {
 	return instances, nil
 }
 
-// doInstancesInventoryWorker gets inventory from cloud for given cloud account.
+// DoResourceInventory gets inventory from cloud for given cloud account.
 func (ec2Cfg *ec2ServiceConfig) DoResourceInventory() error {
+	vpcs, _ := ec2Cfg.getVpcs()
+
 	instances, e := ec2Cfg.getInstances()
 	if e != nil {
 		awsPluginLogger().V(0).Info("error fetching ec2 instances", "account", ec2Cfg.accountName, "error", e)
 	} else {
 		exists := struct{}{}
 		vpcIDs := make(map[string]struct{})
 		instanceIDs := make(map[cloudcommon.InstanceID]*ec2.Instance)
-		vpcNameToID, _ := ec2Cfg.buildMapVpcNameToID()
+		vpcNameToID := ec2Cfg.buildMapVpcNameToID(vpcs)
```

Comment:
Lets discuss how this Maps can be reduced. Please look at the usage of these maps and see if we can get all from 1 internal MAP